client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -108,7 +108,30 @@ public void readerOffline(String readerId, Position lastPosition) {
 
     @Override
     public CompletableFuture<Checkpoint> initiateCheckpoint(String checkpointName, ScheduledExecutorService backgroundExecutor) {
-        synchronizer.updateStateUnconditionally(new CreateCheckpoint(checkpointName));
+
+        String rejectMessage = "rejecting checkpoint request since pending checkpoint reaches max allowed limit";
```

Comment:
Could we unit test the execution of this method by say mocking the state synchronizer? I'd like to have some nit testing in addition to the integration test that is already added below.